echopype/echodata/combine.py

```diff
@@ -435,14 +446,248 @@ def _check_echodata_channels(
+        if not np.isnan(first_times).all():
+            is_descending = (np.diff(first_times) < np.timedelta64(0, "ns")).any()
+
+            if is_descending:
+                raise RuntimeError(
+                    f"The coordinate {time} is not in ascending order for "
+                    f"group {ed_group}, combine cannot be used!"
+                )
+
+
+def _merge_attributes(attributes: List[Dict[str, str]]) -> Dict[str, str]:
+    """
+    Merges a list of attributes dictionary
+
+    Parameters
+    ----------
+    attributes : list of dict
+        List of attributes dictionary
+        E.g. [{'attr1': 'val1'}, {'attr2': 'val2'}, ...]
+
+    Returns
+    -------
+    dict
+        The merged attribute dictionary
+    """
+    merged_dict = {}
+    for attribute in attributes:
+        for key, value in attribute.items():
+            if value == "" and key not in merged_dict:
+                merged_dict[key] = value
+            elif value != "":
+                merged_dict[key] = value
+    return merged_dict
+
+
+def _capture_prov_attrs(
+    attrs_dict: Dict[str, List[Dict[str, str]]], echodata_filenames: List[str], sonar_model: str
+) -> xr.Dataset:
+    """
+    Capture and create provenance dataset,
+    from the combined attribute values.
+
+    Parameters
+    ----------
+    attrs_dict : dict of list
+        Dictionary of attributes for each of the group.
+        E.g. {'Group': [{'attr1': 'val1'}, {'attr2': 'val2'}, ...]}
+    echodata_filenames : list of str
+        The filenames of the echodata objects
+    sonar_model : str
+        The sonar model
+
+    Returns
+    -------
+    xr.Dataset
+        The provenance dataset for all the attribute values
+
+    """
+    index_keys = [ED_GROUP, ED_FILENAME]
+    df_list = []
+    for group, attributes in attrs_dict.items():
+        df = pd.DataFrame.from_records(attributes)
+        df.loc[:, ED_FILENAME] = echodata_filenames
+        df[ED_GROUP] = group
+        df = df.set_index(index_keys)
+        df_list.append(df)
+
+    df_merged = reduce(
+        lambda left, right: pd.merge(left, right, on=index_keys, how="outer"), df_list
+    )
+    prov_ds = df_merged.to_xarray()
+    # Set these provenance as string
+    if sonar_model.lower() != "ek80":
+        # Skip for EK80
+        # TODO: This is right now problematic with array attribute for filter coeff
+        prov_ds = prov_ds.fillna("").astype(str)
+    prov_ds[ED_GROUP] = prov_ds[ED_GROUP].astype(str)
+    prov_ds[ED_FILENAME] = prov_ds[ED_FILENAME].astype(str)
+    return prov_ds
+
+
+def _combine(
+    sonar_model: str,
+    eds: List[EchoData] = [],
+    echodata_filenames: List[str] = [],
+    ed_group_chan_sel: Dict[str, Optional[List[str]]] = {},
+) -> Dict[str, xr.Dataset]:
+    """
+    Combines the echodata objects and export to a dictionary tree.
+
+    Parameters
+    ----------
+    sonar_model : str
+        The sonar model used for all elements in ``eds`
```

Comment:
```suggestion
        The sonar model used for all elements in ``eds``
```

Code suggestion for the above NIT.